src/validator/task.rs

```diff
@@ -88,79 +99,101 @@ enum HandleM3ProposeBundleError {
     InactiveSidechain { sidechain_number: SidechainNumber },
 }
 
-#[derive(Debug, Error)]
+#[fatality(splitable)]
 enum HandleM4VotesError {
     #[error(transparent)]
+    #[fatal]
     DbPut(#[from] db_error::Put),
     #[error(transparent)]
+    #[fatal]
     DbTryGet(#[from] db_error::TryGet),
 }
 
-#[derive(Debug, Error)]
+#[fatality(splitable)]
 enum HandleM4AckBundlesError {
     #[error("Error handling M4 Votes")]
+    #[fatal(forward)]
     Votes(#[from] HandleM4VotesError),
 }
 
-#[derive(Debug, Error)]
+#[fatality(splitable)]
 enum HandleFailedM6IdsError {
     #[error(transparent)]
+    #[fatal]
     DbIter(#[from] db_error::Iter),
     #[error(transparent)]
+    #[fatal]
     DbPut(#[from] db_error::Put),
 }
 
-#[derive(Debug, Error)]
+#[fatality(splitable)]
 enum HandleM5M6Error {
     #[error(transparent)]
+    #[fatal]
     DbPut(#[from] db_error::Put),
     #[error(transparent)]
+    #[fatal]
     DbTryGet(#[from] db_error::TryGet),
     #[error("Invalid M6")]
     InvalidM6,
     #[error("Old Ctip for sidechain {} is unspent", .sidechain_number.0)]
     OldCtipUnspent { sidechain_number: SidechainNumber },
 }
 
-#[derive(Debug, Error)]
+#[fatality(splitable)]
 enum HandleM8Error {
     #[error("BMM request expired")]
     BmmRequestExpired,
     #[error("Cannot include BMM request; not accepted by miners")]
     NotAcceptedByMiners,
 }
 
-#[derive(Debug, Error)]
+#[fatality(splitable)]
 enum ConnectBlockError {
     #[error(transparent)]
+    #[fatal]
     PutBlockInfo(#[from] dbs::block_hash_dbs_error::PutBlockInfo),
     #[error(transparent)]
+    #[fatal]
     DbDelete(#[from] db_error::Delete),
     #[error(transparent)]
+    #[fatal]
     DbFirst(#[from] db_error::First),
     #[error(transparent)]
+    #[fatal]
     DbGet(#[from] db_error::Get),
     #[error(transparent)]
+    #[fatal]
     DbLen(#[from] db_error::Len),
     #[error(transparent)]
+    #[fatal]
     DbPut(#[from] db_error::Put),
     #[error(transparent)]
+    #[fatal]
     DbTryGet(#[from] db_error::TryGet),
     #[error("Error handling failed M6IDs")]
+    #[fatal(forward)]
     FailedM6Ids(#[from] HandleFailedM6IdsError),
     #[error("Error handling failed sidechain proposals")]
+    #[fatal(forward)]
     FailedSidechainProposals(#[from] HandleFailedSidechainProposalsError),
     #[error("Error handling M1 (propose sidechain)")]
+    #[fatal(forward)]
     M1ProposeSidechain(#[from] HandleM1ProposeSidechainError),
     #[error("Error handling M2 (ack sidechain)")]
+    #[fatal(forward)]
     M2AckSidechain(#[from] HandleM2AckSidechainError),
     #[error("Error handling M3 (propose bundle)")]
+    #[fatal(forward)]
     M3ProposeBundle(#[from] HandleM3ProposeBundleError),
     #[error("Error handling M4 (ack bundles)")]
+    #[fatal(forward)]
     M4AckBundles(#[from] HandleM4AckBundlesError),
     #[error("Error handling M5/M6")]
+    #[fatal(forward)]
     M5M6(#[from] HandleM5M6Error),
     #[error("Error handling M8")]
+    #[fatal(forward)]
```

Comment:
It's a bit unfortunate that we have to mark errors as explicitly fatal. In a way I feel like it makes more sense to mark non-fatal errors as such? Seeing as those are going to be far less common